### ISSUE TITLE
quarkus-funqy-http-3.9.3 - ERROR [io.qua.dev.dep.DevUIProcessor] failed to process extension descriptor - class java.lang.String cannot be cast to class java.util.List

### ISSUE THREAD

**Issue by faustool** [quarkusio/quarkus#40140]
### Describe the bug

When running the application with `mvn quarkus:dev`, the error below is printed in the output:

```
2024-04-18 09:13:13,216 ERROR [io.qua.dev.dep.DevUIProcessor] (build-49) Failed to process extension descriptor jar:file:///<path>/.m2/repository/io/quarkus/quarkus-funqy-http/3.9.3/quarkus-funqy-http-3.9.3.jar!/META-INF/quarkus-extension.yaml: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.List (java.lang.String and java.util.List are in module java.base of loader 'bootstrap')
        at io.quarkus.devui.deployment.DevUIProcessor.lambda$getAllExtensions$1(DevUIProcessor.java:496)                                                                                                                                                                                                                                                                                                                                                                                   
        at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeAsPath$0(ClassPathUtils.java:121)                                                                                                                                                                                                                                                                                                                                                                                          
        at io.quarkus.runtime.util.ClassPathUtils.processAsPath(ClassPathUtils.java:154)                                                                                                                                                                                                                                                                                                                                                                                                   
        at io.quarkus.runtime.util.ClassPathUtils.consumeAsPath(ClassPathUtils.java:120)                                                                                                                                                                                                                                                                                                                                                                                                   
        at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:104)
        at io.quarkus.runtime.util.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:85)
        at io.quarkus.devui.deployment.DevUIProcessor.getAllExtensions(DevUIProcessor.java:443)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:483)
```

### Expected behavior

I expected to see the Funqy extensions in the dev-ui/extensions page.

### Actual behavior

No Funqy extension shows in the dev-ui/extensions page.

### How to Reproduce?

Run `mvn quarkus:dev`

### Output of `uname -a` or `ver`

Microsoft Windows 11 Enterprise - 10.0.22631

### Output of `java -version`

OpenJDK Runtime Environment Microsoft-8905927 (build 21.0.2+13-LTS

### Quarkus version or git rev

3.9.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information

This bug happens in all versions released after 3.8.4:

* 3.9.0.CR1
* 3.9.0.CR2
* 3.9.1
* 3.9.2
* 3.9.3
* 3.10.0.CR1

**Comment by bertjan** [comment]
Interesting, I upgraded to Quarkus 3.10 a week ago and everything seemed to work fine. When testing my demo for DevoxxUK this morning, it first complained that I needed a newer Maven version and then the error above occurred. Downgrading the Quarkus versions in the pom "fixed" this. But if I recall correctly, I didn't change anything in my local environment in the mean time. Is this caused by something that Quarkus downloads from the internet during a build? I'm just interested in understanding what was wrong, so no hurry ;-)

**Comment by phillip-kruger** [comment]
The fix for this is not yet in a released version (it should be in 3.11). I can not really comment on the newer Maven version (that seems unrelated).
